tb_plugin/torch_tb_profiler/profiler/multiprocessing.py

```diff
@@ -0,0 +1,27 @@
+# When some filesystems launch, they need to be in a process
+# launched by a certain start method. Here, we provide utility
+# functions to set a start method for only this plugin's
+# subprocesses, without disrupting the start method used in
+# other plugins.
+
+import os
+import multiprocessing as mp
+
+class KinetoContext():
```

Comment:
would it better to add only the following function instead of the static class?

def get_start_method():
    if os.getenv('TORCH_PROFILER_START_METHOD'):
        return os.getenv('TORCH_PROFILER_START_METHOD')
    else:
        return None

then we only need call the mp.get_context by using the returned start method without need to save the global start method?

Reply:
No, this allows us to set the start method dynamically instead of only using the environment flag. Specifically, it lets us call `set_profiler_start_method()` during runtime, so that we can determine the correct start method based on the log directories supplied.

However, this class is still compatible with the environment flag, so it works either way. The reason we need a full class is to store values that were set explicitly by that function.

Reply:
If you need set the start method dynamically, just pass the start_method to the multiprocessing.get_context is fine. I don't think we need maintain the global state

Reply:
I think set the start method by environment is enough. the set_profiler_start_method and TORCH_PROFILER_START_METHOD are do the exactly same thing.